x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/CreateApiKeyResponse.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.security.action;
+
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.time.Instant;
+
+/**
+ * Response for the successful creation of an api key
+ */
+public final class CreateApiKeyResponse extends ActionResponse {
+
+    private String name;
+    private String key;
+    private Instant expiration;
+
+    CreateApiKeyResponse() {}
+
+    public CreateApiKeyResponse(String name, String key, Instant expiration) {
+        this.name = name;
+        this.key = key;
+        this.expiration = expiration;
+    }
+
+    public CreateApiKeyResponse(StreamInput in) throws IOException {
+        this.name = in.readString();
+        this.key = in.readString();
+        this.expiration = in.readOptionalInstant();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getKey() {
+        return key;
+    }
+
+    @Nullable
+    public Instant getExpiration() {
+        return expiration;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
```

Comment:
It bothers me that we call `super.writeTo` but the constructor doesn't have a `super(in)` and `readFrom` doesn't call `super.`
If the base class is ever changed to actually write something, it will break.

It can probably be handled separately, but if we're going to start using Streamable constructors in reponse objects, it would be good to create a super constructor.